### PR TITLE
fundamentals/index (and other) updates

### LINKED ISSUE #2377: fundamentals/index "Fundamentals"
- [ ] Follow-up to #2343 -  Rewrite so that the text corresponds to the articles in the TOC. Change the order of topics in the text and in the TOC as needed.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #2377 

[Internal review topic (link will be added when build completes)](#)

Notes:
* Bit of a long-shot here on the order of topics in the TOC and the order within the fundamentals index page, so we'll need to work on that. Ordering is (loosely) general-to-specific and important-to-less-important.
* I definitely feel the need to add missing fundamentals topic blurbs to the index.
* Added and updated descriptions on fundamentals topics.
* The order of the TOC matches the order of the features described in the fundamentals index page.
* I still wonder: Is it "metapackage" or "meta-package"? ... The :pray: server gods :pray: haven't ruled on this one yet. @Rick-Anderson, can you say??

### CODE REVIEW

File: aspnetcore/fundamentals/index.md [hunk 1]
```diff
@@ -50,87 +50,163 @@ The `UseStartup` method on `WebHostBuilder` specifies the `Startup` class for yo
...
 
+## Environments
+
+Environments, such as "Development" and "Production", are a first-class notion in ASP.NET Core and can be set using environment variables.
+
+For more information, see [Working with Multiple Environments](xref:fundamentals/environments).
+
+## Configuration
+
+ASP.NET Core uses a configuration model based on name-value pairs. The configuration model isn't based on `System.Configuration` or *web.config*. Configuration obtains settings from an ordered set of configuration providers. The built-in configuration providers support a variety of file formats (XML, JSON, INI) and environment variables to enable environment-based configuration. You can also write your own custom configuration providers.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/index.md]: I removed the banded word ~simple~

Comment (guardrex) [reply aspnetcore/fundamentals/index.md]: I like that word tho ... in life I mean. :grinning:

File: aspnetcore/toc.md [hunk 2]
```diff
@@ -73,31 +73,29 @@
 
 # [Fundamentals](fundamentals/index.md)
 ## [Application Startup](fundamentals/startup.md)
+## [Dependency Injection (Services)](fundamentals/dependency-injection.md)
 ## [Middleware](fundamentals/middleware.md)
-## [Working with Static Files](fundamentals/static-files.md)
-## [Routing](fundamentals/routing.md)
-## [URL Rewriting Middleware](fundamentals/url-rewriting.md)
```
Comment (guardrex) [inline aspnetcore/toc.md]: @Rick-Anderson Did you mean to drop this one? It fell out of the TOC on this PR.

Comment (Rick-Anderson) [reply aspnetcore/toc.md]: No - thanks for the heads up, I'll get that fixed ASAP

**Pull request merged**